### PR TITLE
Form and input validation and formatting

### PR DESCRIPTION
**Pull request by yujuncho**
The New Patient form has been updated to reuse the field-renderer component which is currently being used in the New Visit form. A Select component has been added to field-renderer, and the Input, Select, and Radio components now handle validation logic. The Input component now also has the ability to format values. The New Patient form is now utilizing this new validation and formatting logic.

### CODE REVIEW

File: client/public/index.html [hunk 1]
```diff
@@ -123,12 +123,6 @@
       To begin the development, run `npm start` or `yarn start`.
       To create a production bundle, use `npm run build` or `yarn build`.
     -->
-    <script type="text/javascript" src="js/jquery.1.11.1.js"></script> 
-
-   <script src="https://code.jquery.com/jquery-3.5.1.slim.min.js" integrity="sha384-DfXdz2htPH0lsSSs5nCTpuj/zy4C+OGpamoFVy38MVBnE+IbbVYUew+OrCXaRkfj" crossorigin="anonymous"></script>
-    <script src="https://cdn.jsdelivr.net/npm/bootstrap@4.5.3/dist/js/bootstrap.bundle.min.js" integrity="sha384-ho+j7jyWK8fNQe+A12Hb8AhRq26LrZ/JpcUGGOn+Y7RsweNrtN/tE3MoK7ZeZDyx" crossorigin="anonymous"></script>
```
Comment (yujuncho) [inline client/public/index.html:129]: Bring this back into the app

Comment (yujuncho) [reply client/public/index.html:129]: Completed this in the latest commit!

File: client/src/ehr_components/NewPatient.tsx [hunk 2]
```diff
@@ -0,0 +1,245 @@
+import { useHistory } from "react-router-dom";
+import { toastr } from "react-redux-toastr";
+import React, { useState } from "react";
+import Axios from "axios";
+
+import FieldRenderer from "./common_components/field-renderer";
+import { stringify } from "querystring";
+
```
Comment (adbahrani) [inline client/src/ehr_components/NewPatient.tsx:8]: I'm not sure why you added stringify querystring, doesn't look like you use it anyway

File: client/src/ehr_components/NewPatient.tsx [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+    maritalStatus: ""
+  });
+
+  const firebaseURl =
+    "https://idoctorpwa-default-rtdb.firebaseio.com/patients.json";
+
+  const updateFormData = (
+    fieldName: string,
+    event: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>
+  ) => {
```
Comment (adbahrani) [inline client/src/ehr_components/NewPatient.tsx:33]: Nice Addition

File: client/src/ehr_components/NewPatient.tsx [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+      formSubmitted: formSubmitted
+    },
+    {
+      label: "Job",
+      name: "job",
+      type: "Input",
+      inputType: "text",
+      placeholder: "Job",
+      onChange: updateFormData,
+      validateValue: (value: string) => {
```
Comment (adbahrani) [inline client/src/ehr_components/NewPatient.tsx]: Job field would probably be optional. let's try to implement that

File: client/src/ehr_components/NewPatient.tsx [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+    <div className="container">
+      <h2>New Patient</h2>
+      <br />
+      <div className="row  justify-content-center">
+        <form
+          className="w-75   align-content-center"
+          onSubmit={handleClick}
+          noValidate
+        >
+          {fieldsMap.map(field => {
```
Comment (adbahrani) [inline client/src/ehr_components/NewPatient.tsx]: Way to go! great use of field generation

File: client/src/ehr_components/NewPatient.tsx [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+    event: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>
+  ) => {
+    setFormData({
+      ...formData,
+      [fieldName]: event.target.value
+    });
+  };
+
+  let fieldsMap = [
+    {
```
Comment (adbahrani) [inline client/src/ehr_components/NewPatient.tsx]: FYI: as this get bigger, it make sense to move it to another file (e.g new patient fields or constant) just to keep files maintainable

**Review by adbahrani — APPROVED** [review]
Looking good

File: client/src/ehr_components/NewPatient.tsx (hunk 6 above)
Comment (yujuncho) [reply client/src/ehr_components/NewPatient.tsx]: Updated! I moved the fields to a separate file, data/new-patient-fields.ts

File: client/src/ehr_components/NewPatient.tsx (hunk 5 above)
Comment (yujuncho) [reply client/src/ehr_components/NewPatient.tsx]: Thank you! 🙏🏽

File: client/src/ehr_components/NewPatient.tsx (hunk 4 above)
Comment (yujuncho) [reply client/src/ehr_components/NewPatient.tsx]: Updated in the latest commit!

File: client/src/ehr_components/NewPatient.tsx (hunk 2 above)
Comment (yujuncho) [reply client/src/ehr_components/NewPatient.tsx:8]: I'm not sure why that's there either. Regardless, I removed it in recent commits!

**Pull request merged**